File: 프로그래머스/0/120835. 진료 순서 정하기/진료 순서 정하기.py
def solution(emergency):
    answer = {}
    for i,v in enumerate(emergency):
        answer[len(emergency) - i] = v
    numeric_items = {k: v for k, v in answer.items() if isinstance(v, (int, float))}
    sorted_numeric_items = sorted(numeric_items.items(), key=lambda x: x[1])
    sorted_keys = [k for k, v in sorted_numeric_items]

    return sorted_keys

File: 프로그래머스/0/120835. 진료 순서 정하기/진료 순서 정하기.js
function solution(emergency) {
    var answer = [];
    const arr = emergency.reduce((acc, cur, index) => {
        acc[cur] = index + 1
        return acc
    }, {})
    const obj = Object.entries(arr).sort((a, b) =>  Number([0]) - Number(b[0]))
    obj.forEach((v, i) => {
        arr[v[0]] = i + 1
    })
    return emergency.map((v) => arr[v])
}